### ISSUE TITLE
Support changing button background color programmatically

### ISSUE THREAD

**Issue by DSteve595** [ZieIony/Carbon#19]
If, in XML, I have a button with `android:background="@color/green", the button will correctly appear as a green button.
However, there's no way I can see to change this from Java. Calling`setBackgroundColor` does set the color correctly, but the button changes to a plain rectangle and loses its style.

**Comment by DSteve595** [comment]
I've worked around it by adding a PorterDuff filter but it would be very nice to have this option.

**Comment by uneq95** [comment]
When do you want the button color to change? After clicking the button or any time in the activity??

**Comment by DSteve595** [comment]
I'm using it in a list adapter. Some items should have a green button and
some should be red.

**Comment by ZieIony** [comment]
This is because RippleDrawable is a decorator. Changing background color clears the decoration. For now, as a workaround you can set RippleDrawable once again by yourself. To change color:

```
view.setBackgroundDrawable(new RippleDrawable(yourRippleColor, new ColorDrawable(yourBackgroundColor));
```

I'm marking this issue as a bug.

**Comment by ZieIony** [comment]
fixed in 0.6.3
